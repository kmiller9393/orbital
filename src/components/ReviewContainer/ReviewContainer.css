.flash-card {
  background-color: burlywood;
  height: 40px;
}

.red-card {
  background-color: red;
  height: 40px;
}
