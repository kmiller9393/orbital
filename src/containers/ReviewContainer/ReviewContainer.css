.review-container-image {
  height: 55px;
  margin-top: 1.5rem;
  width: 125px; }

.week-title {
  color: #f7f7f7;
  font-family: 'Chalkboard SE', 'Times New Roman', serif; }

.button-container {
  display: flex;
  justify-content: space-evenly; }

.review-header-container {
  padding-right: 6.5rem; }

.back-arrow {
  height: 40px;
  padding: 0rem 4rem 0.5rem 0rem; }

.review-container-button {
  background-color: #009c92;
  border-color: #009c92;
  border-radius: 5px;
  color: #f7f7f7;
  font-family: 'Chalkboard SE', 'Times New Roman', serif;
  font-size: 1.2rem;
  height: 45px;
  margin-top: 1rem;
  width: 110px; }
