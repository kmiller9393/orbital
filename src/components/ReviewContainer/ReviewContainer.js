import React from 'react';
import CardOne from '../../containers/CardOne/CardOne';
import { connect } from 'react-redux';
import './ReviewContainer.css';

const ReviewContainer = props => {
  let facts = props.facts.slice(0, 9);
  const setOneQuestions = facts.map((subject, index) => (
    <CardOne subject={subject.name} answer={subject.definition} key={index} />
  ));

  return (
    <div>
      <h2>Orbital</h2>
      <h2>Week 1 Review</h2>
      <div>{setOneQuestions}</div>
    </div>
  );
};

export const mapStateToProps = state => ({
  facts: state.facts
});

export default connect(
  mapStateToProps,
  null
)(ReviewContainer);
