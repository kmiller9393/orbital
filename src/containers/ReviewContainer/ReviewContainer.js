import React, { Component } from 'react';
import CardOne from '../CardOne/CardOne';
import { NavLink, withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import orbital from '../../images/orbital-header.svg';
import './ReviewContainer.css';

class ReviewContainer extends Component {
  constructor() {
    super();
    this.state = {
      page: 0
    };
  }

  navigateCard = e => {
    const { page } = this.state;
    const { history, facts } = this.props;
    let newPage = page;

    e.target.name === 'next' ? newPage++ : newPage--;

    if (newPage < 0 || newPage > 8) return;
    this.setState({ page: newPage });
    history.replace(`/review-one/${facts[newPage].name}`);
  };

  render() {
    const { page } = this.state;
    const { facts } = this.props;

    return (
      <div>
        <NavLink to="/">
          <img className="review-container-image" src={orbital} alt="Orbital" />
        </NavLink>
        <h2 className="week-title">Week 1 Review</h2>
        <CardOne {...facts[page]} key={page} />
        <div className="button-container">
          <button
            className="review-container-button"
            onClick={this.navigateCard}
            name="previous"
          >
            Previous
          </button>
          <button
            className="review-container-button"
            onClick={this.navigateCard}
            name="next"
          >
            Next
          </button>
        </div>
      </div>
    );
  }
}

export const mapStateToProps = state => ({
  facts: state.glossary
});

export default withRouter(
  connect(
    mapStateToProps,
    null
  )(ReviewContainer)
);
