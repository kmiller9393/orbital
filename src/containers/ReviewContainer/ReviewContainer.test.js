import ReviewContainer, { mapStateToProps } from './ReviewContainer';

describe('ReviewContainer', () => {
  describe('ReviewContainer component', () => {});

  describe('mapStateToProps', () => {
    it('should return an array of glossary objects', () => {
      const glossary = {
        name: 'Asteroid',
        definition:
          'A small solar system object composed mostly of rock. Many of these objects orbit the Sun between Mars and Jupiter. Their sizes range anywhere from 33 feet (10 meters) in diameter to less than 620 miles (1,000 kilometers). The largest known asteroid, Ceres, has a diameter of 579 miles (926 kilometers).',
        id: 0
      };

      const mockState = {
        glossary
      };
      const expected = {
        glossary: {
          name: 'Asteroid',
          definition:
            'A small solar system object composed mostly of rock. Many of these objects orbit the Sun between Mars and Jupiter. Their sizes range anywhere from 33 feet (10 meters) in diameter to less than 620 miles (1,000 kilometers). The largest known asteroid, Ceres, has a diameter of 579 miles (926 kilometers).',
          id: 0
        }
      };

      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(expected);
    });
  });
});
