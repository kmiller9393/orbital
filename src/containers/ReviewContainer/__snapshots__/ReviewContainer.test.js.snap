// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewContainer ReviewContainer component should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ReviewContainer
    glossary={
        Array [
            Object {
              "definition": "A small solar system object composed mostly of rock. Many of these objects orbit the Sun between Mars and Jupiter. Their sizes range anywhere from 33 feet (10 meters) in diameter to less than 620 miles (1,000 kilometers). The largest known asteroid, Ceres, has a diameter of 579 miles (926 kilometers).",
              "id": 0,
              "name": "Asteroid",
            },
          ]
    }
    history={
        Object {
            "location": Object {
              "pathname": "/review-one",
            },
            "replace": [Function],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <NavLink
          activeClassName="active"
          aria-current="page"
          to="/"
>
          <img
                    alt="Orbital"
                    className="review-container-image"
                    src="orbital-header.svg"
          />
</NavLink>,
        <h2
          className="week-title"
>
          Weekly Review
</h2>,
        <ReviewCard
          definition="A small solar system object composed mostly of rock. Many of these objects orbit the Sun between Mars and Jupiter. Their sizes range anywhere from 33 feet (10 meters) in diameter to less than 620 miles (1,000 kilometers). The largest known asteroid, Ceres, has a diameter of 579 miles (926 kilometers)."
          id={0}
          name="Asteroid"
/>,
        <div
          className="button-container"
>
          <button
                    className="review-container-button"
                    name="previous"
                    onClick={[Function]}
          >
                    Previous
          </button>
          <button
                    className="review-container-button"
                    name="next"
                    onClick={[Function]}
          >
                    Next
          </button>
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": <img
            alt="Orbital"
            className="review-container-image"
            src="orbital-header.svg"
/>,
          "to": "/",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Orbital",
            "className": "review-container-image",
            "src": "orbital-header.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Weekly Review",
          "className": "week-title",
        },
        "ref": null,
        "rendered": "Weekly Review",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "class",
        "props": Object {
          "definition": "A small solar system object composed mostly of rock. Many of these objects orbit the Sun between Mars and Jupiter. Their sizes range anywhere from 33 feet (10 meters) in diameter to less than 620 miles (1,000 kilometers). The largest known asteroid, Ceres, has a diameter of 579 miles (926 kilometers).",
          "id": 0,
          "name": "Asteroid",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              className="review-container-button"
              name="previous"
              onClick={[Function]}
>
              Previous
</button>,
            <button
              className="review-container-button"
              name="next"
              onClick={[Function]}
>
              Next
</button>,
          ],
          "className": "button-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Previous",
              "className": "review-container-button",
              "name": "previous",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Previous",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Next",
              "className": "review-container-button",
              "name": "next",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Next",
            "type": "button",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NavLink
            activeClassName="active"
            aria-current="page"
            to="/"
>
            <img
                        alt="Orbital"
                        className="review-container-image"
                        src="orbital-header.svg"
            />
</NavLink>,
          <h2
            className="week-title"
>
            Weekly Review
</h2>,
          <ReviewCard
            definition="A small solar system object composed mostly of rock. Many of these objects orbit the Sun between Mars and Jupiter. Their sizes range anywhere from 33 feet (10 meters) in diameter to less than 620 miles (1,000 kilometers). The largest known asteroid, Ceres, has a diameter of 579 miles (926 kilometers)."
            id={0}
            name="Asteroid"
/>,
          <div
            className="button-container"
>
            <button
                        className="review-container-button"
                        name="previous"
                        onClick={[Function]}
            >
                        Previous
            </button>
            <button
                        className="review-container-button"
                        name="next"
                        onClick={[Function]}
            >
                        Next
            </button>
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": <img
              alt="Orbital"
              className="review-container-image"
              src="orbital-header.svg"
/>,
            "to": "/",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "Orbital",
              "className": "review-container-image",
              "src": "orbital-header.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Weekly Review",
            "className": "week-title",
          },
          "ref": null,
          "rendered": "Weekly Review",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "class",
          "props": Object {
            "definition": "A small solar system object composed mostly of rock. Many of these objects orbit the Sun between Mars and Jupiter. Their sizes range anywhere from 33 feet (10 meters) in diameter to less than 620 miles (1,000 kilometers). The largest known asteroid, Ceres, has a diameter of 579 miles (926 kilometers).",
            "id": 0,
            "name": "Asteroid",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                className="review-container-button"
                name="previous"
                onClick={[Function]}
>
                Previous
</button>,
              <button
                className="review-container-button"
                name="next"
                onClick={[Function]}
>
                Next
</button>,
            ],
            "className": "button-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Previous",
                "className": "review-container-button",
                "name": "previous",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Previous",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Next",
                "className": "review-container-button",
                "name": "next",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Next",
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
